### ISSUE TITLE
What is the appropriate training loss?

### ISSUE THREAD

**Issue by dreamercv** [lightly-ai/lightly#928]
I use light to train my own data, there are a total of 30w pieces of data（the data captured by the driving recorder）。

The parameters I set are epoch=1000, batch=128, optimizer=sgd，after training for 300 epochs, when I connected the self-supervised model to the downstream task, I found that the effect of the downstream task was very poor

 how do I set the hyperparameters? for example epoch ？batch size ？ optimizer？What is the approximate range of loss?In other words, how do I judge whether a self-supervised model is good or bad? （Of the 350,000 pieces of data, only 50,000 are labeled.）

**Comment by MalteEbner** [comment]
Hey @dreamercv ,

### Regarding hyperparameters:
These benchmarks show that usually, more epochs and higher benchmarks are better: https://docs.lightly.ai/getting_started/benchmarks.html

We did not benchmark different optimisers so far, best take the one wich is recommended for the SSL model you use. Look at the reproduction scripts linked in the docs. The loss depends on the optimiser, model and dataset, but it is usually in the region of 2 to 4.

The most important hyperparameters are usually the augmentations you choose. E.g. random flips are ok for images, but not for any text. Color jitter may work well for some datasets, but not for others. Think about what augmentations should not change the embedding and which should not.

### Regarding the downstream task

> when I connected the self-supervised model to the downstream task, I found that the effect of the downstream task was very poor

Could you elaborate here? What do you mean by "connected"? What model did you use on the downstream task. If/how did you use fine-tuning? What does you downstream task look like (Classification or something else?)

**Comment by dreamercv** [comment]
I use the backbone of yolox for self-supervised training. After training, I use the p3, p4, and p5 layers of pafpn as the output, and then add several layers of convolution after p3, p4, and p5 as my downstream task network (specific task is the keypoint detection where the vehicle meets the ground).

How to train downstream tasks？
![eIIC5A0wat](https://user-images.githubusercontent.com/50707593/192454567-822be47d-64e7-4b1c-a427-a9c6e40e556b.png)

**Comment by MalteEbner** [comment]
Most SSL models learn to distinguish images, thus their embeddings encode properties of the full image and neglect details. This makes SSL pretraining great for classification tasks, but worse for tasks focusing on details like object detection and keypoint detection. This explains why the pretaining does not work.

I see some possibilities here to fix that:
- Train you SSL model not only on full images but rather on images cut out by the bounding boxes. E.g. cut your example image with the bus by the blue bounding box. Then train on these object crops. This will allow the SSL to encode more street details and neglect things like the sky or surrounding buildings.
- Do not freeze your backbone (do NOT locke the network of pafpn)
- use a SSL model different from SimCLR that learns the details better. Asking @guarin which models might be more suitable.

**Comment by guarin** [comment]
As @MalteEbner suggested it is often best to not completely freeze the backbone for finetuning on object detection. Regarding the different SSL models you can, for example, find the following table in the [Efficient Visual Pretraining with Contrastive Detection](https://arxiv.org/abs/2103.10957) paper (Table 2):
<img width="551" alt="Screenshot 2022-09-28 at 10 37 19" src="https://user-images.githubusercontent.com/43336610/192732621-a8140778-fc14-4c68-a216-13c00bbe754f.png">

There clearly exist some better methods but you should also get a reasonable accuracy with SimCLR.

**Comment by dreamercv** [comment]
In my work, there are currently 1 million pictures, of which 400,000 pictures are marked with rectangular frames of people and vehicles, 400,000 pictures are marked with road information such as lane lines, and the remaining 200,000 pictures are marked with vehicles and ground. information on key points. However, these data are all models trained separately, including the lane line detection model, the vehicle grounding point model, and the human and vehicle detection model. It can be seen from the above that 1 million pieces of data have not been fully used. Now I want to use the SSL method to train a backbone separately (all 1 million pictures participate in the training), and then connect to specific downstream tasks (vehicle detection, grounding point detection) , lane line detection, etc.); the specific method is as follows: engineer A trains the backbone separately and maintains and updates it; engineer X trains the downstream tasks alone (backbone locks); thus constructing a multi-task model. How can I train this multi-task model?
I watched the video below and wanted to try this method?
The purpose of this is that the data can be fully utilized, and even unlabeled data can be utilized.
https://www.bilibili.com/video/BV1Ca411g7AY/?is_story_h5=false&p=1&share_from=ugc&share_medium=android&share_plat=android&share_session_id=4111fca6-261f-40bd-940a-ac492670144d&share_source=COPY&share_tag=s_i&timestamp=1663393237&unique_k=PzlPIjI&vd_source=bf7f18773f27ff026b2ce0a3aaa2f291

![img_v2_47f46681-588d-441d-8bb9-469db71bb8dg](https://user-images.githubusercontent.com/50707593/192969968-74be1bde-2c1a-4130-b24c-0b92c22a5dac.jpg)
